File: typescript/src/renderer.ts
// import { newRectangle, Rectangle } from "./shapes/rectangle.js";
import {Shape} from './shapes/shape'

function newRenderer(shape: Shape) {
    return {
        draw() {
            const area: number = shape.computeArea()
            const name: string = shape.getName()
            console.log("Shape drawn:" + name + "\nIts area is " + area)
        }
    }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    // public Rectangle rectangle;
    public Shape shape;
    
    // Renderer(Rectangle rectangle) {
    //     this.rectangle = rectangle;
    // }
    Renderer(Shape shape) {
        // this.rectangle = rectangle;
        this.shape = shape;
    }

    void draw() {
        // double area = rectangle.getArea();
        String shapename = shape.getName();
        double area = shape.getArea();

        // assume implementation

        System.out.println("Shape printed: " + shapename + "\nIts area is " + area);
    }
}
